src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -127,13 +132,20 @@ export class SearchInterceptor {
     options?: ISearchOptions
   ): Promise<IKibanaSearchResponse> {
     const { abortSignal, sessionId, ...requestOptions } = options || {};
+    const { isRestore, isStored, legacyHitsTotal, strategy } = {
+      ...requestOptions,
+      ...this.deps.session.getSearchOptions(sessionId),
```

Comment:
`getSearchOptions(sessionId)` returns `null` in some cases. In cases when it returns `null` we shouldn't include `sessionId` `isRestore` `isStored` as part of the request. 

This is how it worked before but with this change the behavior is different

Reply:
Done.